subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/settings/views/SubsSettingBuyView.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.subscriptions.impl.settings.views
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import com.duckduckgo.anvil.annotations.InjectWith
+import com.duckduckgo.di.scopes.ViewScope
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+import com.duckduckgo.navigation.api.GlobalActivityStarter
+import com.duckduckgo.subscriptions.impl.SubscriptionsActivity.Companion.SubscriptionsScreenWithEmptyParams
+import com.duckduckgo.subscriptions.impl.databinding.ViewSettingsBuyBinding
+import com.duckduckgo.subscriptions.impl.settings.views.SubsSettingBuyViewModel.Command
+import com.duckduckgo.subscriptions.impl.settings.views.SubsSettingBuyViewModel.Command.OpenBuyScreen
+import com.duckduckgo.subscriptions.impl.settings.views.SubsSettingBuyViewModel.Factory
+import dagger.android.support.AndroidSupportInjection
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(ViewScope::class)
+class SubsSettingBuyView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : FrameLayout(context, attrs, defStyle) {
+
+    @Inject
+    lateinit var viewModelFactory: Factory
+
+    @Inject
+    lateinit var globalActivityStarter: GlobalActivityStarter
+
+    private var coroutineScope: CoroutineScope? = null
+
+    private val binding: ViewSettingsBuyBinding by viewBinding()
+
+    private val viewModel: SubsSettingBuyViewModel by lazy {
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, viewModelFactory)[SubsSettingBuyViewModel::class.java]
+    }
+
+    override fun onAttachedToWindow() {
+        AndroidSupportInjection.inject(this)
+        super.onAttachedToWindow()
+
+        binding.buy.setClickListener {
+            viewModel.onBuyClicked()
+        }
+
+        @SuppressLint("NoHardcodedCoroutineDispatcher")
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+
+        viewModel.commands()
```

Comment:
I'd also create a `ConflatedJog` and `+=` it. Later cancel it as well in `onDetachedFromWindow`